tests/test_generate_polygons_cli.py

```diff
@@ -0,0 +1,69 @@
+import math
+import os
+from pathlib import Path
+
+import fsspec
+import geopandas as gpd
+import pytest
+from click.testing import CliRunner
+
+from deafrica_waterbodies.cli.generate_polygons import generate_polygons
+
+# Test directory.
+HERE = Path(__file__).parent.resolve()
+TEST_WATERBODY = os.path.join(HERE, "data", "sm9rtw98n.geojson")
+TEST_OUTPUT_DIRECTORY = HERE / "test_outputs"
+
+
+@pytest.fixture
+def runner():
+    return CliRunner(echo_stdin=True)
+
+
+def test_generate_polygons(runner, capsys: pytest.CaptureFixture):
+    aoi_vector_file = TEST_WATERBODY
+    tile_size_factor = 1
+    num_workers = 8
+    detection_threshold = 0.1
+    extent_threshold = 0.05
+    min_valid_observations = 60
+    # raster_processing_plugin_name = "raster_processing_filtering"
+    output_directory = TEST_OUTPUT_DIRECTORY
+    min_polygon_size = 4500
+    max_polygon_size = math.inf
+    length_threshold_km = 150
```

Comment:
Suggest removing filtering by length from this step and only doing this for conflux. Length attribute should still be calculated.